manufacturing-ai-suite/wind-turbine-anomaly-detection/simulator/mqtt-publisher/publisher.py

```diff
@@ -72,43 +72,44 @@ def stream_csv(mqttc, topic, subsample, sampling_rate, filename):
     """
     Stream the csv file
     """
-    target_start_time = time.time()
-    row_count = 0
-    row_served = 0
-    jencoder = json.JSONEncoder()
-
-    print("\nMQTT Topic - {}\nSubsample - {}\n \
-          Sampling Rate - {}\nFilename - {}\n".format(topic,
-                                                      subsample,
-                                                      sampling_rate,
-                                                      filename))
+    loop = os.getenv("LOOP", "true").lower()
+    while True:
+        start_time = time.time()
+        row_served = 0
+        jencoder = json.JSONEncoder()
 
-    with open(filename, 'r') as fileobject:
-        tick = g_tick(float(subsample) / float(sampling_rate))
+        print(f"\nMQTT Topic - {topic}\nSubsample - {subsample}\nSampling Rate - {sampling_rate}\nFilename - {filename}\n")
 
-        for row in fileobject:
+        with open(filename, 'r') as fileobject:
+            tick = g_tick(float(subsample) / float(sampling_rate))
 
-            row = [x for x in row.split(',') if x not in (' \n', ' \r\n')]
-            if re.match(r'^-?\d+', row[0]) is None:
-                continue
+            for row in fileobject:
+                row = [x.strip() for x in row.split(',') if x.strip()]
+                if not row or not re.match(r'^-?\d+', row[0]):
+                    continue
 
-            row_count += 1
-            if (subsample > 1) and (row_count % subsample) != 1:
-                continue
-            row_served += 1
+                if subsample > 1 and (row_served % subsample) != 0:
+                    row_served += 1
+                    continue
 
-            values = [float(x) for x in row]
-            msg = jencoder.encode({'grid_active_power': values[0], 'wind_speed': values[1]})
-            print("Publishing message ", msg)
-            mqttc.publish(topic, msg)
+                try:
+                    values = [float(x) for x in row]
+                    msg = jencoder.encode({'grid_active_power': values[0], 'wind_speed': values[1]})
+                    print("Publishing message", msg)
+                    mqttc.publish(topic, msg)
+                except (ValueError, IndexError):
+                    continue
```

Comment:
Let's log the error message so we get to know what data points is having problem, ideally we shouldn't be having any problem